packages/x-data-grid-premium/src/hooks/features/aggregation/createAggregationLookup.ts

```diff
@@ -62,39 +87,46 @@ const getGroupAggregatedValue = (
         columnAggregationRules.aggregationFunction as GridAggregationFunction;
       const field = aggregatedField;
 
-      if (aggregatedValues[j] === undefined) {
-        aggregatedValues[j] = {
-          aggregatedField,
-          values: [],
-        };
-      }
-
+      let value;
       if (typeof aggregationFunction.getCellValue === 'function') {
-        aggregatedValues[j].values.push(aggregationFunction.getCellValue({ row }));
+        value = aggregationFunction.getCellValue({ field, row });
+      } else if (isPivotActive) {
+        // Since we know that pivoted fields are flat, we can use the row directly, and save lots of processing time
```

Comment:
That's a good assumption since all `valueGetter`s are being called when building the pivoted set of rows 👍🏻